packages/pricing/src/services/currency.ts

```diff
@@ -0,0 +1,107 @@
+import { Context, DAL, FindConfig, PricingTypes } from "@medusajs/types"
+import {
+  InjectManager,
+  InjectTransactionManager,
+  MedusaContext,
+  ModulesSdkUtils,
+  retrieveEntity,
+} from "@medusajs/utils"
+import { Currency } from "@models"
+import { CurrencyRepository } from "@repositories"
+
+import { doNotForceTransaction, shouldForceTransaction } from "@medusajs/utils"
+
+type InjectedDependencies = {
+  currencyRepository: DAL.RepositoryService
+}
+
+export default class CurrencyService<TEntity extends Currency = Currency> {
+  protected readonly currencyRepository_: DAL.RepositoryService
+
+  constructor({ currencyRepository }: InjectedDependencies) {
+    this.currencyRepository_ = currencyRepository
+  }
+
+  @InjectManager("currencyRepository_")
+  async retrieve(
+    currencyId: string,
```

Comment:
thought: Should this not be `currencyCode` as this is the primary key in the table?